LogMonitor/src/LogMonitor/JsonProcessor.cpp

```diff
@@ -0,0 +1,284 @@
+    std::wstringstream wss;
+    wss << wif.rdbuf();
+
+    // Close the file stream
+    wif.close();
+
+    // Convert wstring buffer to string (UTF-8) for further processing
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
+    std::string jsonString = converter.to_bytes(wss.str());
+
+    return jsonString;
+}
+
+// Function to handle EventLog type
+void handleEventLog(
+    _In_ const boost::json::value& source,
+    _In_ AttributesMap& Attributes,
+    _Inout_ std::vector<std::shared_ptr<LogSource>>& Sources
+) {
+    bool startAtOldest = source.as_object().at("startAtOldestRecord").as_bool();
+    bool multiLine = source.at("eventFormatMultiLine").as_bool();
+    std::string customLogFormat = source.as_object().at("customLogFormat").as_string().c_str();
+
+    Attributes[JSON_TAG_START_AT_OLDEST_RECORD] = reinterpret_cast<void*>(
+        new std::wstring(startAtOldest ? L"true" : L"false")
+        );
+    Attributes[JSON_TAG_FORMAT_MULTILINE] = reinterpret_cast<void*>(new std::wstring(multiLine ? L"true" : L"false"));
+    Attributes[JSON_TAG_CUSTOM_LOG_FORMAT] = reinterpret_cast<void*>(
+        new std::wstring(Utility::string_to_wstring(customLogFormat))
+        );
+
+    // Process channels if they exist
+    if (source.as_object().contains("channels")) {
+        std::vector<EventLogChannel>* channels = new std::vector<EventLogChannel>();
+        for (const auto& channel : source.as_object().at("channels").as_array()) {
+            std::string name = channel.as_object().at("name").as_string().c_str();
+            std::string levelString = channel.as_object().at("level").as_string().c_str();
+
+            EventLogChannel channel(Utility::string_to_wstring(name), EventChannelLogLevel::Error);
+            // Set the level based on the levelString
+            if (channel.SetLevelByString(Utility::string_to_wstring(levelString))) {
+                // Successfully set the level
+            }
+            else {
+                logWriter.TraceError(
+                    Utility::FormatString(
+                        L"Invalid level string: %lu",
+                        Utility::string_to_wstring(levelString)
+                    ).c_str()
+                );
+            }
+            // Add the channel to the vector
+            channels->push_back(channel);
+        }
+        Attributes[JSON_TAG_CHANNELS] = channels;
+    }
+
+    std::shared_ptr<SourceEventLog> sourceEventLog = std::make_shared< SourceEventLog>();
+
+    //
+    // Fill the new EventLog source object, with its attributes
+    //
+    if (!SourceEventLog::Unwrap(Attributes, *sourceEventLog))
+    {
+        logWriter.TraceError(
+            L"Error parsing configuration file. "
+            L"Invalid EventLog source (it must have a non-empty 'channels')"
+        );
+    }
+
+    Sources.push_back(std::reinterpret_pointer_cast<LogSource>(std::move(sourceEventLog)));
+}
+
+// Function to handle File type
+void handleFile(
```

Comment:
nit: could qualify this to `handleFileLog`

Reply:
Function renamed.